framework/Distributions.py

```diff
@@ -1925,6 +1948,31 @@ def rvs(self):
       self.pot = np.resize(self.pot, self.pot.size - 1)
     return rvsValue
 
+  def selectedRvs(self,discardedElems):
+    """
+      Return a random state of the distribution without discardedElems
+      @ In, discardedElems, np array, list of discarded elements
+      @ Out, rvsValue, float, the random state
+    """
+    if self.nPoints is None:
+      self.xArray   = np.arange(self.lowerBound,self.upperBound+1)
+    else:
+      self.xArray   = np.linspace(self.lowerBound,self.upperBound,self.nPoints)
+
+    self.xArray = np.setdiff1d(self.xArray,discardedElems)
+
+    self.pdfArray = 1/self.xArray.size * np.ones(self.xArray.size)
+    paramsDict={}
+    paramsDict['xAxis'] = self.xArray
+    paramsDict['pAxis'] = self.pdfArray
+    paramsDict['strategy'] = self.strategy
+
+    self.tempUniformDiscrete = UniformDiscrete()
+    self.tempUniformDiscrete.initializeFromDict(paramsDict)
+
+    rvsValue = self.tempUniformDiscrete.rvs()
+    return rvsValue
```

Comment:
This does not need to be fixed, but I think it would be quicker (computationally) and less lines to just sample from the self.xArray directly.

Reply:
The decision to go through a UniformDiscrete() distribution was to make sure we had control of RNG.
It is indeed more complicated but I thought it would be neater.